actors/multisig/tests/multisig_actor_test.rs

```diff
@@ -1847,23 +1847,112 @@ fn test_lock_balance_checks_preconditions() {
+                code: ExitCode::OK,
+            },
+            TestCase {
+                desc: "fail to set threshold to zero",
+                initial_threshold: 2,
+                set_threshold: 0,
+                code: ExitCode::USR_ILLEGAL_ARGUMENT,
+            },
+            TestCase {
+                desc: "fail to set threshold above number of signers",
+                initial_threshold: 2,
+                set_threshold: 4,
+                code: ExitCode::USR_ILLEGAL_ARGUMENT,
+            },
+        ];
+
+        for tc in test_cases {
+            let mut rt = construct_runtime(msig);
+            let h = util::ActorHarness::new();
+            let signers = vec![anne, bob, chuck];
+            h.construct_and_verify(&mut rt, tc.initial_threshold, 0, 0, signers);
+            rt.set_caller(*MULTISIG_ACTOR_CODE_ID, msig);
+            let ret = h.change_num_approvals_threshold(&mut rt, tc.set_threshold);
+            match tc.code {
+                ExitCode::OK => {
+                    assert_eq!(RawBytes::default(), ret.unwrap());
+                    let st: State = rt.get_state();
+                    assert_eq!(tc.set_threshold, st.num_approvals_threshold);
+                }
+                _ => {
+                    assert_eq!(
+                        tc.code,
+                        ret.expect_err("change threshold return expected to be actor error")
+                            .exit_code()
+                    );
+                }
+            }
+        }
+    }
+
+    #[test]
+    fn tx_can_be_reapproved_and_execed_after_threshold_lowered() {
+        let msig = Address::new_id(100);
+        let anne = Address::new_id(101);
+        let bob = Address::new_id(102);
+        let chuck = Address::new_id(103);
+
+        let mut rt = construct_runtime(msig);
+        let h = util::ActorHarness::new();
+        let signers = vec![anne, bob, chuck];
+        let num_approvals = 2;
+
+        h.construct_and_verify(&mut rt, num_approvals, 0, 0, signers);
+
+        // anne proposes tx id 0
+        let fake_method = 42;
+        let send_value = TokenAmount::from(10u8);
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, anne);
+        let proposal_hash =
+            h.propose_ok(&mut rt, chuck, send_value.clone(), fake_method, RawBytes::default());
+
+        // lower approval threshold, tx is technically approved, but will not be executed yet
+        rt.set_caller(*MULTISIG_ACTOR_CODE_ID, msig);
+        h.change_num_approvals_threshold(&mut rt, 1).unwrap();
+
+        // anne may re-approve causing tx to be exected
+        rt.expect_send(
+            chuck,
+            fake_method,
+            RawBytes::default(),
+            send_value.clone(),
+            RawBytes::default(),
+            ExitCode::OK,
+        );
+        rt.set_balance(send_value);
+        rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, anne);
+        h.approve_ok(&mut rt, TxnID(0), proposal_hash);
+        h.assert_transactions(&rt, vec![]);
```

Comment:
should we check the state at the end, like in the reference test?

Reply:
We need to eventually but we've been deferring actually writing the state checks so I'm planning to add in the state check calls once those are written.  Side note -- writing the actor state checks might be one of the more interesting subtasks so you're welcome to take a look at it and assign yourself if you want to.